fuel_gauge: Fix desired current/voltage units

aaronemassey opened the pull request:
The desired current/voltage properties make use of milliamps/volts while the present current/voltage properties make use of microamps/volts.

Fix the desired current/voltage properties to be consistent with the present current/voltage properties where they're most likely to be used with.

Added to https://github.com/zephyrproject-rtos/zephyr/issues/61818 API improvements issue. Units should be documents directly in the properties. Relying on documentation isn't great.

aaronemassey committed:
fuel_gauge: Fix desired current/voltage units

The desired current/voltage properties make use of milliamps/volts while
the present current/voltage properties make use of microamps/volts.

Fix the desired current/voltage properties to be consistent with the
present current/voltage properties where they're most likely to be used
with.

Signed-off-by: Aaron Massey <aaronmassey@google.com>

kartben reviewed (dismissed):
Thanks!
Btw I've noticed that it looks like the comment on line 29 in fuel_gauge.h re: alphabetically ordering the properties is not really followed. You will probably want to either reorder everything at some point, or just remove the comment?

aaronemassey committed:
fuel_gauge: Remove keep alphabetized comment

The fuel gauge header asks to alphabetize the properties in order to keep
them organized and easy to read. However, this is difficult to enforce
without adding an adhoc script for just this header. In addition, there
aren't so many properties such that it's difficult to parse unsorted. Since
the alphabetization is already not followed and not likely to be easily
followed in the future, remove this ask from the header.

Remove "keep properties alphabetized" from the header.

Signed-off-by: Aaron Massey <aaronmassey@google.com>

aaronemassey commented:
@kartben 

> Thanks! Btw I've noticed that it looks like the comment on line 29 in fuel_gauge.h re: alphabetically ordering the properties is not really followed. You will probably want to either reorder everything at some point, or just remove the comment?

Yeah it's a difficult requirement to follow with the docs inlined with the properties. So it's not as easy as just calling some sort routine on that region of text.

Removing that comment. Thank you!

kartben reviewed (approved):
I meant to +1 earlier, not just comment :)
